app/src/main/java/io/github/omochice/pinosu/domain/usecase/FetchRelayListUseCase.kt

```
@@ -0,0 +1,75 @@
+  }
+
+  override suspend fun invoke(npubPubkey: String): Result<List<RelayConfig>> {
+    if (!npubPubkey.startsWith("npub1") || npubPubkey.length != 63) {
+      return Result.failure(
+          IllegalArgumentException("Invalid npub format: must be Bech32-encoded (npub1...)"))
+    }
+
+    val hexPubkey =
+        try {
+          Bech32.npubToHex(npubPubkey)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)